### PR TITLE
Feat(compiler_base): add compiler_base_session

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #115

#### 2. What is the scope of this PR (e.g. component or file name):

compiler_base_session

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

add struct `Session` and trait `SessionDiagnostic` in compiler_base_session.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: compiler_base/session/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+///         Ok(diag)
+///     }
+/// }
+///
+/// // 3. The diagnostic of MyError will display "error" on terminal.
+/// // For more information about diagnositc displaying, see doc in `compiler_base_error`.
+/// ```
+///
+/// TODO(zongz): [WIP] This should not be implemented manually. Instead, use
+/// `#[derive(SessionDiagnostic)]` -- see [rustc_macros::SessionDiagnostic].
```
Comment (Peefy) [inline compiler_base/session/src/lib.rs:82]: see [compiler_base::session::SessionDiagnostic] ?

**Review by Peefy — APPROVED** [review]
LGTM

**Review by ldxdl — APPROVED** [review]
LGTM

File: compiler_base/session/src/lib.rs (hunk 1 above)
Comment (zong-zhe) [reply compiler_base/session/src/lib.rs]: fixed

**Pull request merged**